File: Benchmarks/Core/Types/Color.cs
namespace Benchmarks.Core.Types
{
    public struct Color(byte r, byte g, byte b, byte a)
    {
        public byte R { get; set; } = r;
        public byte G { get; set; } = g;
        public byte B { get; set; } = b;
        public byte A { get; set; } = a;

        public static readonly Color White = new(255, 255, 255, 255);
        public static readonly Color Black = new(0, 0, 0, 255);
        public static readonly Color Red = new(255, 0, 0, 255);
        public static readonly Color Green = new(0, 255, 0, 255);
        public static readonly Color Blue = new(0, 0, 255, 255);
    }
}

File: Benchmarks_FSharp/Core/Types/Color.fs
namespace Benchmarks.Core.Types

[<Struct>]
type Color =
    val mutable R: byte
    val mutable G: byte
    val mutable B: byte
    val mutable A: byte
    new(r, g, b, a) = { R = r; G = g; B = b; A = a }

    static member White = Color(255uy, 255uy, 255uy, 255uy)
    static member Black = Color(0uy, 0uy, 0uy, 255uy)
    static member Red = Color(255uy, 0uy, 0uy, 255uy)
    static member Green = Color(0uy, 255uy, 0uy, 255uy)
    static member Blue = Color(0uy, 0uy, 255uy, 255uy)
